src/main/java/net/dries007/tfc/client/model/PlantBlockModel.java

```diff
@@ -0,0 +1,309 @@
+                start = start + 0.5f;
+            }
+            start = (start + random.nextFloat(-randomScale, randomScale)) % 1;
+            return getModelFromCalendar(start, start + plant.getBloomingEnd(), start + plant.getSeedingEnd(), start + plant.getDyingEnd(),
+                start + plant.getDormantEnd(), start + plant.getSproutingEnd(), plant.getStartTime(), plant.getEndTime(), randomScale > 0.25f);
+        }
+    }
+
+    private BakedModel getModelFromCalendar()
+    {
+        return getModelFromCalendar(0.4f, 0.6f, 0.75f, 0.9f, 1.1f, 1.25f, 0, 0, false);
+    }
+
+    private BakedModel getModelFromCalendar(float bloomingStart, float bloomingEnd, float seedingEnd, float dyingEnd, float dormantEnd, float sproutingEnd, int startTime, int endTime, boolean nonDormant)
+    {
+        final float timeOfYear = Calendars.CLIENT.getCalendarFractionOfYear();
+        final float adjustedTimeOfYear = timeOfYear < bloomingStart ? timeOfYear + 1f : timeOfYear;
+
+        if (adjustedTimeOfYear < bloomingEnd)
+        {
+            if (startTime != endTime)
+            {
+                return getModelByDayTime(startTime, endTime);
+            }
+            assert bloomingBakedModel != null;
+            return bloomingBakedModel;
+        }
+        else if (adjustedTimeOfYear < seedingEnd)
+        {
+            assert seedingBakedModel != null;
+            return seedingBakedModel;
+        }
+        else if (adjustedTimeOfYear < dyingEnd)
+        {
+            if (nonDormant)
+            {
+                assert buddingBakedModel != null;
+                return buddingBakedModel;
+            }
+            assert dyingBakedModel != null;
+            return dyingBakedModel;
+        }
+        if (adjustedTimeOfYear < dormantEnd)
+        {
+            if (nonDormant)
+            {
+                if (startTime != endTime)
+                {
+                    return getModelByDayTime(startTime, endTime);
+                }
+                assert bloomingBakedModel != null;
+                return bloomingBakedModel;
+            }
+            assert dormantBakedModel != null;
+            return dormantBakedModel;
+        }
+        else if (adjustedTimeOfYear < sproutingEnd)
+        {
+            if (nonDormant)
+            {
+                assert seedingBakedModel != null;
+                return seedingBakedModel;
+            }
+            assert sproutingBakedModel != null;
+            return sproutingBakedModel;
+        }
+        else
+        {
+            assert buddingBakedModel != null;
+            return buddingBakedModel;
+        }
+    }
+
+    public BakedModel getModelByDayTime(float startTime, float endTime)
+    {
+        final int dayTime = (int) Calendars.CLIENT.getCalendarDayTime();
+        if ((endTime < dayTime && dayTime < startTime) || (startTime < endTime && (dayTime < startTime || endTime < dayTime)))
```

Comment:
Yeah, it does leave the potential for these plants to get slightly out of sync with their actual schedules, but from what I was seeing in testing random ticks update chunks often enough that it should be okay. Can always revert this later if we decide we don't like it.

Reply:
Maybe a nitpick, but it might be simpler if instead of `getFractionOfDay() * 24000` we just use `getFractionOfDay() * 24` and call it `hours` in the plant definition. Since we're already deviating from vanilla daytime here, it's probably best not to confuse that with `level.dayTime()`